PCR Next.js port

luke-rt opened the pull request:
- Backend changes: add JWT authentication as an authentication class so isAuthenticated permission class accepts both Platform session id and JWT tokens
- Update /api/review/autocomplete route to exclude semester in search results

luke-rt committed:
Nextjs initialization

luke-rt committed:
Set up tailwind, cva, shadcn

luke-rt committed:
styling, fonts, began landing page

rydersitcawich committed:
added about page

rydersitcawich committed:
faq page

shiva-menta committed:
Add Fetch Locations Management Command

luke-rt committed:
styling, scorebox, api helpers

luke-rt committed:
Nextjs route rewrites

luke-rt committed:
updated Next.js version, rewrites

luke-rt committed:
Small fixes, began JWT auth migration

luke-rt committed:
JWT tokens working with oidc somewhat working

luke-rt committed:
Removed old auth stuff

luke-rt committed:
Rollback to tailwind 3 for compatability

luke-rt committed:
Progress with next-auth configuration

luke-rt committed:
Made some progress with nextauth and middleware

luke-rt committed:
Trying without nextauth

luke-rt committed:
JWT token logic working, need to cryptographically verify

luke-rt committed:
Create test route and setup JWK

luke-rt committed:
Resolved dev environment bugs

luke-rt committed:
Removed duplicate dependencies

luke-rt committed:
Got auth middleware working >:)

sritanmotati committed:
rename of token

luke-rt committed:
Began working on design

luke-rt committed:
Set up debounced course search

luke-rt committed:
Searchbar in header

luke-rt committed:
Bump dependencies, upgrade tailwind

luke-rt committed:
Fix expired token bug and cache JWKs

luke-rt committed:
Use shadcn for searchbar

luke-rt committed:
Reconfigure Shadcn, finish searchbar styling

luke-rt committed:
Nits

luke-rt committed:
Finished fuzzy searchbar/autocomplete

luke-rt committed:
Progress with static rendering review data

el-agua committed:
deleted the semester field

el-agua committed:
Revert "deleted the semester field"

This reverts commit e2172ecf747f830fcec105865ef670c3740e5462.

el-agua committed:
Removed the semester url redirect

luke-rt committed:
Change searchbar style in header

luke-rt committed:
Lint :(

luke-rt committed:
Merge branch 'master' into review-nextjs

luke-rt committed:
Rollback postgres version to align w master

luke-rt committed:
Fixed frontend middleware, started oidc backend middleware

luke-rt committed:
JWK verification finished in middleware

luke-rt committed:
Caching JWKs client with thread locking

luke-rt committed:
jk made it a custom auth permission class

luke-rt committed:
Broke something

luke-rt committed:
Allow AnonymousUser in JWT auth and working fetching on frontend

luke-rt committed:
Linting

luke-rt committed:
Linting

luke-rt committed:
Frontend progress

luke-rt committed:
Merge branch 'master' into review-nextjs

luke-rt committed:
Thank you copilot

luke-rt committed:
Linting

luke-rt committed:
Linting

luke-rt committed:
Update docker

el-agua commented on backend/PennCourses/authentication/jwt.py:
```diff
@@ -0,0 +1,62 @@
+import logging
```
Consider rewriting this with the asyncio library, using asyncio.create_task() and asyncio.sleep(). We use it in our webhook management commands.

el-agua reviewed (commented):
Where are the helper functions in authentication/jwt.py being called?

luke-rt commented on backend/review/views.py:
```diff
@@ -854,3 +855,18 @@ def join_depts(depts):
+        },
+    )
+)
+def test_jwt(request):
```
Yes, will remove later

luke-rt commented on backend/PennCourses/settings/base.py:
```diff
@@ -14,6 +14,10 @@
 
 import boto3
 import dj_database_url
+import dotenv
```
Probably not but rn we only have a test credential anyways

luke-rt committed:
Rewrite to asyncio

luke-rt committed:
Update workflow hash

luke-rt committed:
Merge branch 'master' into review-nextjs

luke-rt committed:
Linting

luke-rt committed:
Hide searchbar on redirect and unfocus

luke-rt committed:
Merge branch 'master' into review-nextjs

jamesdoh0109 committed:
add appDir true for next config

jamesdoh0109 committed:
limit search results to avoid search speed/crashing

jamesdoh0109 committed:
add minor edits + fix tab spacing

luke-rt committed:
Merge pull request #727 from pennlabs/james/review-nextjs

James/review nextjs

jamesdoh0109 committed:
edit middleware config paths to match new route structure

jamesdoh0109 committed:
fix redirect url from / to entered path

jamesdoh0109 committed:
Merge pull request #728 from pennlabs/james/fix-redirect-after-log-in

fix redirect url from / to entered path

luke-rt committed:
Heavy refactoring

luke-rt committed:
Indexed db stale-while-revalidate

luke-rt committed:
Small changes

luke-rt committed:
Add null check for data

jamesdoh0109 committed:
fix invalid jwt token due to missing cryptography package

luke-rt committed:
Merge pull request #730 from pennlabs/james/fix-invalid-jwt-token

fix invalid jwt token due to missing cryptography package

luke-rt committed:
Moved JWT to Celery

luke-rt committed:
Put review pages in header

luke-rt committed:
Remove command autoscroll

luke-rt committed:
Comment

jamesdoh0109 committed:
Merge remote-tracking branch 'origin/master' into review-nextjs

luke-rt committed:
Moved async jwt refresh to read() function

luke-rt committed:
Merge pull request #729 from pennlabs/luke/review-nextjs-search

Instant Search using fuzzysort

luke-rt committed:
Lint

luke-rt committed:
Oops

jamesdoh0109 reviewed (approved):
😈

jamesdoh0109 committed:
install cryptography per uv migration

jamesdoh0109 committed:
remove print statements

jamesdoh0109 committed:
Merge pull request #732 from pennlabs/james/install-cryptography-following-uv-migration

James/install cryptography following uv migration

luke-rt committed:
Fix test cases (#733)

Fix test cases to match new autocomplete route. Fix new JWTAuthentication strategy to match spec

luke-rt committed:
Rename to review-2.0
